sormas-app/app/src/main/java/de/symeda/sormas/app/immunization/vaccination/VaccinationNewActivity.java

```diff
@@ -146,8 +196,17 @@ protected void onPreExecute() {
 
 			@Override
 			public void doInBackground(TaskResultHolder resultHolder) throws DaoException {
-				final Vaccination savedVaccination = DatabaseHelper.getVaccinationDao().saveAndSnapshot(Vaccination);
-				final Immunization immunization = DatabaseHelper.getImmunizationDao().queryUuid(Vaccination.getImmunization().getUuid());
+
+				Immunization newImmunization = buildImmunizationFromRootEntity();
```

Comment:
The whole creation logic should only be executed when the immunization feature is reduced; otherwise the saving should be done as usual because in that case we already have an immunization (since we can only create vaccinations from inside immunizations anyway).

Reply:
I updated the logic so only in case of Reduced vaccination the whole creation logic will be triggered.